exp-android-sdk/src/main/java/com/scala/exp/android/sdk/model/AbstractModel.java

```diff
@@ -61,9 +64,24 @@ public String toString() {
         return this.properties.toString();
     }
 
-
-    public void fling(IChannel channel, Map payload){
-        channel.fling(payload);
+    /**
+     * Get Channel with options
+     * @param options
+     * @return
+     */
+    public IChannel getChannel(Map options){
+        IChannel channel = null;
+        if(options!=null && !options.isEmpty()){
+            boolean consumer = false;
+            boolean system = false;
+            if(options.containsKey(CONSUMER)){
+                 consumer = (Boolean) options.get(CONSUMER);
+            }else if(options.containsKey(SYSTEM)){
+                 system = (Boolean) options.get(SYSTEM);
+            }
+            channel = Exp.getChannel(getUuid(),system,consumer);
```

Comment:
We also need the getChannel() for Data to override this.  It doesn't use uuid for the channel name.  See: https://github.com/ScalaInc/exp-js-sdk/blob/master/src/API.js#L363